opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/WildcardQuery.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+package org.opensearch.sql.opensearch.storage.script.filter.lucene.relevance;
+
+import com.google.common.collect.ImmutableMap;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.index.query.WildcardQueryBuilder;
+
+/**
+ * Lucene query that builds wildcard query.
+ */
+public class WildcardQuery extends SingleFieldQuery<WildcardQueryBuilder> {
+  /**
+   *  Default constructor for WildcardQuery configures how RelevanceQuery.build() handles
+   * named arguments.
+   */
+  public WildcardQuery() {
+    super(ImmutableMap.<String,
+        QueryBuilderStep<WildcardQueryBuilder>>builder()
+        .put("boost", (b, v) -> b.boost(Float.parseFloat(v.stringValue())))
+        .put("case_insensitive", (b, v) -> b.caseInsensitive(Boolean.parseBoolean(v.stringValue())))
+        .put("rewrite", (b, v) -> b.rewrite(v.stringValue()))
```

Comment:
Can you
1. Move this to `FunctionParameterRepository`
2. Use existing checkers from it (or add new ones)
For example, for `boost`:
```java
    .put("boost", (b, v) -> b.boost(convertFloatValue(v, "boost")))
```

Reply:
Thanks. I'll have to rebase 2.x for this. I will wait for https://github.com/opensearch-project/sql/pull/1010 to get merged so that I can implement this change and resolve merge conflicts.

Reply:
Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/156/commits/e05e5c5097fb4c69926952d08a7dde1aa1e00da1